### PR TITLE
fix(plugin-m): move to 'waitForCatalog'

### PR DESCRIPTION
**Pull request by mwegman**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by mwegman** [comment]
The tests for these 2 packages already existed and were failing. Now they are succeeding.

File: packages/node_modules/@webex/plugin-memberships/src/memberships.js [hunk 1]
```diff
@@ -255,7 +254,7 @@ const Memberships = WebexPlugin.extend({
     };
 
     return ensureMyIdIsAvailable(this.webex)
-      .then(() => ensureConvoAvailability(this.webex)
+      .then(() => this.webex.internal.services.waitForCatalog('postauth')
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/plugin-memberships/src/memberships.js:257]: This should work well, note that if you're looking for a specific service, the `webex.internal.services.waitForService()` method could be used as well. Both should work based on the usecase I'm seeing though.

File: packages/node_modules/@webex/common/src/wdm-register.js [hunk 2]
```diff
@@ -1,18 +0,0 @@
...
- * @param {Object} webex
- * @returns {Promise}
- */
-export default async function ensureConvoAvailability(webex) {
-  if (webex.internal.device.services.conversationServiceUrl) {
-    return Promise.resolve();
-  }
-
-  return webex.internal.device.register();
-}
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/common/src/wdm-register.js:17]: Good find, as this method/data should no longer exist in the `devices` plugin.

**Comment by InteractiveTimmy** [comment]
Be sure to remove any tests from the plugin that involve this additional method if there are any. I don't believe this requires any tests unless you'd like to confirm that it calls `waitForCatalog()` at some point.

**Pull request merged**